src/NServiceBus.Extensions.DependencyInjection/ServiceCollectionAdapter.cs

```diff
@@ -73,13 +76,13 @@ public void RegisterSingleton<T>(T instance)
             RegisterSingleton(typeof(T), instance);
         }
 
-        void RegisterInterfaces(Type component)
+        void RegisterInterfaces(Type component, ServiceLifetime lifetime)
         {
             var interfaces = component.GetInterfaces();
             foreach (var serviceType in interfaces)
             {
                 // see https://andrewlock.net/how-to-register-a-service-with-multiple-interfaces-for-in-asp-net-core-di/
-                serviceCollection.Add(new ServiceDescriptor(serviceType, sp => sp.GetService(component), ServiceLifetime.Transient));
+                serviceCollection.Add(new ServiceDescriptor(serviceType, sp => sp.GetService(component), lifetime));
```

Comment:
this seems to be only necessary to be compatible with Castle.Windsor, but as long as the lifetimes of the forwarding registration and the "main" registration are equal, that shouldn't be an issue.

Reply:
@timbussmann besides castle windsor i think there is one more issue which i tried to explain [here](https://github.com/Particular/NServiceBus.Extensions.DependencyInjection/issues/20#issuecomment-603252464)